influxql/iterator.gen.go.tmpl

```diff
@@ -497,33 +531,33 @@ func new{{$k.Name}}FillIterator(input {{$k.Name}}Iterator, expr Expr, opt Iterat
 		auxFields = make([]interface{}, len(opt.Aux))
 	}
 
-	itr := &{{$k.name}}FillIterator{
+	return &{{$k.name}}FillIterator{
 		input:     newBuf{{$k.Name}}Iterator(input),
 		startTime: startTime,
 		endTime:   endTime,
 		auxFields: auxFields,
 		opt:       opt,
 	}
-
-	p := itr.input.peek()
-	if p != nil {
-		itr.window.name, itr.window.tags = p.Name, p.Tags
-		itr.window.time = itr.startTime
-	} else {
-		if opt.Ascending {
-			itr.window.time = itr.endTime + 1
-		} else {
-			itr.window.time = itr.endTime - 1
-		}
-	}
-	return itr
```

Comment:
It's moved into `Next()` to avoid a call to `peek()` inside of the initializer. That way, we reduce the time that a query is uninterruptible. It also has the benefit of removing the else statement which never really worked correctly, but existed there to attempt to convince `Next()` to return nil. Now, `Next()` can just return nil instead of trying to find some weird way for these functions to communicate with each other.